homeWork5/lesson1/Program.cs

```diff
@@ -0,0 +1,449 @@
+
+//--
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator })-- = { rNumbers1-- }");
+Console.WriteLine($"({ rNumbers2.Numerator } / { rNumbers2.Denumenator })-- = { rNumbers2-- }");
+Console.WriteLine($"({ rNumbers3.Numerator } / { rNumbers3.Denumenator })-- = { rNumbers3-- }");
+Console.ReadKey();
+
+public class RatinalNumbers
+{
+    public RatinalNumbers(int numerator, int denumerator)
+    {
+        _numerator = numerator;
+        _denumerator = denumerator;
+    }
+
+    private int _numerator;
+    private int _denumerator;
+
+    public int Numerator
+    {
+        get => _numerator;
+        set => _numerator = value;      
+    }
+
+    public int Denumenator
+    {
+        get => _denumerator;
+        set => _denumerator = value;
+    }
+
+/// <summary>
+/// Переопределенный метод Equals() 
+/// </summary>
+/// <param name="obj">object</param>
+/// <returns>bool</returns>
+    public override bool Equals(object? obj)
+    {
+        if (obj is not RatinalNumbers rnumbers)
+        {
+           return false;
+        }
+        return Numerator == rnumbers.Numerator &&
+               Denumenator == rnumbers.Denumenator;
+    }
+
+/// <summary>
+/// Переопределенный метод GetHashCode()
+/// </summary>
+/// <returns>int</returns>
+    public override int GetHashCode()
+    {
+        int hashcode = _numerator.GetHashCode();
+        hashcode = 31 * hashcode + _denumerator.GetHashCode();
+        return hashcode;
+    }
+
+/// <summary>
+/// Переопределенный метод ToString()
+/// </summary>
+/// <returns>string</returns>
+    public override string ToString()
+    {
+        return $"{ _numerator }/{ _denumerator }";
+    }
+
+    /// <summary>
+    /// Переопределенный оператор приведения float
+    /// </summary>
+    /// <param name="rnumbers">RatinalNumbers</param>
+    public static explicit operator float(RatinalNumbers rnumbers)
+    {
+        return (float)rnumbers.Numerator / rnumbers.Denumenator;
+    }
+
+    /// <summary>
+    /// Переопределенный оператор приведения int
+    /// </summary>
+    /// <param name="rnumbers">RatinalNumbers</param>
+    public static explicit operator int(RatinalNumbers rnumbers)
+    {
+        return (int)rnumbers.Numerator / rnumbers.Denumenator;
+    }
+
+    /// <summary>
+    /// Переопределенный оператор ==
+    /// </summary>
+    /// <param name="rnumbers1">RatinalNumbers</param>
+    /// <param name="rnumbers2">RatinalNumbers</param>
+    /// <returns>bool</returns>
+    public static bool operator ==(RatinalNumbers rnumbers1, RatinalNumbers rnumbers2)
+    {
+        if (ReferenceEquals(rnumbers1, rnumbers2))
+        {
+            return true;
+        }
+            
+        if (ReferenceEquals(rnumbers1, null)) 
+        {
+            return false;
+        }
+            
+        if (ReferenceEquals(rnumbers2, null))
+        {
+            return false;
+        }
+        return rnumbers1.Equals(rnumbers2);

```

Comment:
Логика этого оператора реализуется через вызов Equals
```suggestion
        return Equals(rnumbers1, rnumbers2);
```